File: src/main/java/gg/climb/lolobjects/RiotId.java
package gg.climb.lolobjects;

import org.immutables.value.Value;

/**
 * Representation of Riot's Id numbers. You can obtain a RiotId through
 * {@link ImmutableRiotId#of(String)}. However, you likely should not be building this class outside
 * of database access and downloading from Riot's servers.
 */
@SuppressWarnings("unused") // Actually used type parameter
@Value.Immutable(builder = false)
public abstract class RiotId<T> {
  @Value.Parameter
  public abstract String getId();
}

File: src/main/scala/gg/climb/lolobjects/RiotId.scala
package gg.climb.lolobjects

class RiotId[T](val id: String){
	override def toString = s"id:$id"
  override def equals(other: Any): Boolean = {
    other match {
      case riotId: RiotId[T] => this.id.equals(riotId.id)
      case _ => false
    }
  }
}

object RiotId {
	def apply[T](id : String) = new RiotId[T](id)
}
